File: SAMPLES/PYTHON/test_strings.py
# test_strings.py - Test TODO 20: CHAINES (str) OPERATIONS AVANCEES
# Compile : PYTHON86 test_strings.py test_strings.asm

# Concatenation
a = "hello" + " " + "world"
print(a)

# Repetition
b = "ab" * 3
print(b)

# Longueur
n = len("hello")
print(n)

# Index positif
s = "abcdef"
c = s[0]
print(c)

# Index negatif
d = s[-1]
print(d)

# Slice
e = s[1:4]
print(e)

# upper / lower
f = "hello"
g = f.upper()
print(g)

h = "WORLD"
i = h.lower()
print(i)

# strip
j = "  hi  "
k = j.strip()
print(k)

# find
s2 = "hello world"
idx = s2.find("world")
print(idx)

idx2 = s2.find("xyz")
print(idx2)

# startswith / endswith
print(s2.startswith("hello"))
print(s2.endswith("world"))
print(s2.startswith("world"))

# replace
r = s2.replace("world", "python")
print(r)

# count
s3 = "abcabcabc"
print(s3.count("abc"))

# in operator
if "lo" in "hello":
    print(1)
if "xyz" in "hello":
    print(0)

# chr / ord (deja implementes, mais testons)
x = chr(65)
print(x)
y = ord("A")
print(y)

File: SAMPLES/LUA/test_strings.lua
-- test_strings.lua : test chaines et bibliotheque string (TODO 19)

-- Test 1 : string.len
local s = "hello"
print(string.len(s))

-- Test 2 : #s (longueur)
print(#s)

-- Test 3 : string.upper / string.lower
print(string.upper("hello"))
print(string.lower("WORLD"))

-- Test 4 : string.sub
print(string.sub("abcdef", 2, 4))

-- Test 5 : string.byte
print(string.byte("A"))

-- Test 6 : string.char
print(string.char(65))

-- Test 7 : string.rep
print(string.rep("ab", 3))

-- Test 8 : string.reverse
print(string.reverse("hello"))

-- Test 9 : string.find
print(string.find("hello world", "world"))

-- Test 10 : string.find not found
print(string.find("hello", "xyz"))

-- Test 11 : string.gsub
print(string.gsub("hello world", "world", "lua"))

-- Test 12 : string.format with %d
print(string.format("x=%d", 42))

-- Test 13 : string.format with %s
print(string.format("name=%s", "lua"))
